src/Axe/Controllers/CompilerController.cs

```diff
@@ -0,0 +1,97 @@
+                    SyntaxTree syntaxTree = CSharpSyntaxTree.ParseText(codeBlock.Code);
+
+                    MetadataReference[] references = new MetadataReference[]
+                    {
+                    MetadataReference.CreateFromFile(typeof(object).GetTypeInfo().Assembly.Location),
+                    MetadataReference.CreateFromFile(typeof(Hashtable).GetTypeInfo().Assembly.Location)
+                        // MetadataReference.CreateFromFile(typeof(object).Assembly.Location),
+                        // MetadataReference.CreateFromFile(typeof(Enumerable).Assembly.Location)
+                    };
+
+                    CSharpCompilation compilation = CSharpCompilation.Create(
+                        assemblyName: assemblyName,
+                        syntaxTrees: new[] { syntaxTree },
+                        references: references,
+                        options: new CSharpCompilationOptions(OutputKind.DynamicallyLinkedLibrary));
+
+                    using (var stream = new MemoryStream())
+                    {
+                        EmitResult result = compilation.Emit(stream);
+                        if (!result.Success)
+                        {
+                            IEnumerable<Diagnostic> failures = result.Diagnostics.Where(diagnostic =>
+                                diagnostic.IsWarningAsError ||
+                                diagnostic.Severity == DiagnosticSeverity.Error);
+
+                            foreach (Diagnostic diagnostic in failures)
+                            {
+                                Console.Error.WriteLine("{0}: {1}", diagnostic.Id, diagnostic.GetMessage());
+                            }
+                            ViewData["StatusOfExecuting"] = "Error";
+                        }
+                        else
+                        {
+                            stream.Seek(0, SeekOrigin.Begin);
+
+                            AssemblyLoadContext context = AssemblyLoadContext.Default;
+                            Assembly assembly = context.LoadFromStream(stream);
+                            MethodInfo method = assembly.GetType("RoslynCompileSample.Writer").GetMethod("Write");
+                            Object writerClass = assembly.CreateInstance("RoslynCompileSample.Writer");
+                            Object output = method.Invoke(writerClass, new Object[] { new int[] { 6, 3, 1, 59334, 232, 3, -1 } });
+
+                            Console.WriteLine("Method returned {0}", output);
+                            CodeBlock codeBlock1 = this.context.CodeBlock.First(cb => cb.Id == codeBlock.Id);
+                            var answer = (codeBlock1.Output == output.ToString());
+
+                            ViewData["StatusOfExecuting"] = answer ? "Ok" : "Output isn't correct";
+                        }
+                    }
+                }
+            }
+            return View("Index", null);
```

Comment:
after attempt to execute code page should not loose all user input. all typed codeBlock.Code should be displayed with additional "Ok" or "Error" status